File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.py
def solution(n, m):
    answer = []
    divisor = 0
    
    for i in range(1, min(n, m)+1):
        if n % i == 0 and m % i == 0:
            divisor = i
    answer.append(divisor)
    
    for i in range(max(n, m), (n * m) + 1):
        if i % n == 0 and i % m == 0:
            answer.append(i)
            break
    
    
    return answer

File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.java
class Solution {
    public int[] solution(int n, int m) {
        int[] answer = new int[2];
        int max = n>=m?n:m;
        int min = n>=m?m:n;
        int div = 0;
        int mul = max;
        
        for(int i = 1; i <= min; i++){
            if(max % i == 0 && min % i == 0){
                div = i;
            }
        }
        answer[0] = div;
        
        while(true){
            if(mul % min == 0){
                break;
            }
            
            mul += max;
            
        }
        
        answer[1] = mul;
        
        return answer;
    }
}